source/python-best-practise.rst

```diff
@@ -0,0 +1,498 @@
+
+    *   Brugeren bør ikke have nogen grund til at teste koden.
+    *   Pakke bør ikke have en masse overflødig funktionalitet med, herunder test-data.
+
+*   Dokumentations-materiale ud over README-filen og eventuelle små-filer til dokumentation, ligger i sin egen mappe af samme årsag som med kildekoden: Dokumentationen skal også bygges og udgives.
+
+*   Alle andre mapper og filer er konfigurationer, scripts og andet til brug primært for udviklere samt for brugeren, der skal installere pakken ud fra arkivet.
+
+
+Udviklerens synsvinkel
+----------------------
+
+| Som udvikler
+| skal du have nogle byggematerialer,
+| der gør det muligt at bygge pakken.
+
+.. Som udvikler har du to primære modtagere:
+
+.. *   dig selv og andre udviklere på projektet
+.. *   brugeren / modtageren.
+
+Filer, som understøtter alt arbejde med kode, dokumentation, etc. ligger som hovedregel i arkivets rod eller i mapper herunder, som grupperer efter formål eller værktøj.
+
+Disse filer og mapper er kun til brug af udvikleren og bør være adskilt fra kildekode, test-funktionalitet, dokumentation og andre slutprodukter.
+
+
+Opsæt versionsstyring
+---------------------
+
+Vi går ikke her ind i detaljerne med at oprette et nyt Git-arkiv til pakken, men skitserer i det følgende skridt til at oprette og arbejde med Git og GitHub.
+
+**Start et nyt arkiv**
+
+*   Opret et nyt Git-arkiv på Github, som skal fungere som den officielle placering af din python-pakke.
+
+    *   Sig ja til at oprette README, LICENSE og ikke mindst en standard ``.gitignore``-fil til Python.
+
+*   I GitHub, opret en *fork* af det nye Git-arkiv til din egen GitHub-bruger.
+
+*   Kopiér SSH-adressen til din fork'ede version af arkivet.
+
+*   I dit udviklingsmiljø [eksempelvis din SIT-PC eller], klon din fork med SSH-adressen, så du kan arbejde lokalt med ændringerne.
+
+
+**Ændringer**
+
+Når du laver ændringer, kan processen være som følger:
+
+*Lokalt*
+
+*   Opret en ny branch til dine ændringer.
+*   Check den nye branch ud og lav dine ændringer.
+*   Skub ændringerne til din fork.
+
+*På Github*
+
+*   Opret et Pull-request til det centrale arkiv.
+*   Hvis ændringerne kan accepteres, så lav et merge af ændringerne til det centrale arkiv.
+
+
+**Første ændringer**
+
+Begynd med at tilpasse README-filen, som GitHub automatisk oprettede for dig. Den bliver dit mest læste dokument og vises automatisk, når man tilgår arkivet på GitHub. Forklar som minimum læseren:
+
+*   hvad projektet går ud på, hvem projektet er til for, og hvordan det skaber værdi (eksistensgrundlag),
+*   hvordan man kommer igang med at bruge pakken,
+*   hvordan man kan bidrage til projektet,
+*   hvordan projektet vedligeholdes,
+
+
+**I det følgende, bliver alle ændringer foretaget lokalt, med mindre andet er angivet.**
+
+
+Python-miljø-opsætning
+----------------------
+
+Vi starter med at installere et nyt udviklingsmiljø med ``mamba``, så vi har en isoleret projekt-opsætning, hvor vi kan fastholde afhængigheder.
```

Comment:
Her, eller et andet sted i nærheden, synes jeg godt det kan give mening at skrive en sætning eller to om man er så godt som garanteret at have ens udviklingsmiljøer på forskellige maskiner. Og dermed undgår en masse bøvl med forskellige versioner af afhængigheder osv. Simpelthen at skære ud i pap hvorfor det her er en god ide - det tror jeg mange har svært ved at forstå uden at have oplevet det på egen krop.

Reply:
Ja, jeg har det i en note, hvor overskriften for dette her inkluderer ordet reproducérbare miljøer.